GCTC-NTGC/gc-digital-talent
# Feature - 2642 - Download CSV

Resolves #2642 

## Summary

This allows admins to download a CSV from the admin users table.

## What This Does

- Changes the download button text to indicate CSV and updates the disable logic
- Adds a new `<DownloadCsv />` link component that implements `react-csv`
- Adds a new hook to massage an `Applicant` to `Record<string, string>` for `react-csv`
- Replaces the existing `<Button />` with the new  `<DownloadCsv />` passing the appropriate data
- Adds a test that implements `cy-verify-downloads` to test that a file does indeed get downloaded

## Note

Currently, our government employee type `localizedConstants.ts` helper returns a `React.ReactNode` which results in `[Object, Object]` appearing in the CSV. I added a simple version of this specifically for CSV's, just want some confirmation this was the right way to go about this.

## Testing

1. Build the admin application `npm run production --workspace=admin`
2. Login as an admin at `/admin`
3. Navigate to `/admin/users/`
4. Select any number of users with the checkbox in the first column
5. Click "Download CSV" button in the table footer
6. Confirm the following:
    - File is downloaded
    - It can be opened easily with Microsoft Excel
    - It contains the expected data from the users you selected in step 4
